librz/bin/p/bin_z64.c

```diff
@@ -79,14 +78,36 @@ static bool check_buffer(RzBuffer *b) {
 	return !memcmp(magic, "\x80\x37\x12\x40", 4);
 }
 
+static bool n64_read_firm_hdr(RzBuffer *buf, N64Header *hdr) {
+	ut64 offset = 0;
+	return rz_buf_read_ble8_offset(buf, &offset, &hdr->x1, false) &&
+		rz_buf_read_ble8_offset(buf, &offset, &hdr->x2, false) &&
+		rz_buf_read_ble8_offset(buf, &offset, &hdr->x3, false) &&
+		rz_buf_read_ble8_offset(buf, &offset, &hdr->x4, false) &&
+		rz_buf_read_le32_offset(buf, &offset, &hdr->ClockRate) &&
+		rz_buf_read_le32_offset(buf, &offset, &hdr->BootAddress) &&
+		rz_buf_read_le32_offset(buf, &offset, &hdr->Release) &&
+		rz_buf_read_le32_offset(buf, &offset, &hdr->CRC1) &&
+		rz_buf_read_le32_offset(buf, &offset, &hdr->CRC2) &&
+		rz_buf_read_le64_offset(buf, &offset, &hdr->UNK1) &&
+		rz_buf_read_offset(buf, &offset, (ut8 *)hdr->Name, sizeof(hdr->Name)) &&
+		rz_buf_read_le32_offset(buf, &offset, &hdr->UNK2) &&
+		rz_buf_read_le16_offset(buf, &offset, &hdr->UNK3) &&
+		rz_buf_read_ble8_offset(buf, &offset, &hdr->UNK4, false) &&
+		rz_buf_read_ble8_offset(buf, &offset, &hdr->ManufacturerID, false) &&
+		rz_buf_read_le16_offset(buf, &offset, &hdr->CartridgeID) &&
+		rz_buf_read_ble8_offset(buf, &offset, (ut8 *)&hdr->CountryCode, false) &&
+		rz_buf_read_ble8_offset(buf, &offset, &hdr->UNK5, false);
```

Comment:
You intermix little and big endian reading here. You need to just use the `ble` version of the functinos

Reply:
Nah, you need to either read big or little endian depending to the file ending. See the link above.